cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -624,11 +636,48 @@ public Secret generateBrokersSecret() {
         return portList;
     }
 
+    /**
+     * Fill the with volumes, persistent volume claims and related volume mount paths for the storage
+     * It's called recursively on the related inner volumes if the storage is of {@link Storage#TYPE_JBOD} type
+     *
+     * @param storage the Storage instance from which building volumes, persistent volume claims and related volume mount paths
+     */
+    private void setDataVolumesClaimsAndMountPaths(Storage storage) {
+        if (storage != null) {
+            Integer id;
+            if (storage instanceof EphemeralStorage) {
+                id = ((EphemeralStorage) storage).getId();
+            } else if (storage instanceof PersistentClaimStorage) {
+                id = ((PersistentClaimStorage) storage).getId();
+            } else if (storage instanceof JbodStorage) {
+                for (SingleVolumeStorage volume : ((JbodStorage) storage).getVolumes()) {
+                    if (volume.getId() == null)
+                        throw new InvalidResourceException("Volumes under JBOD storage type have to have id property");
+                    setDataVolumesClaimsAndMountPaths(volume);
+                }
+                return;
+            } else {
+                throw new InvalidResourceException("The declared storage is not supported");
```

Comment:
This isn't even possible is it? So I think this could be `IllegalStateException`, because if it happened it would indicate a programming error in the operator rather than anything the user could fix.

Reply:
Agree